tests/ui/const-generics/forbid-non-structural_match-types.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0741]: `C` must be annotated with `#[derive(PartialEq, Eq)]` to be used as the type of a const parameter
-  --> $DIR/forbid-non-structural_match-types.rs:11:19
+error[E0741]: `C` must be annotated with `#[derive(ConstParamTy)]` to be used as the type of a const generic parameter
```

Comment:
I could also special case this message to suggest `#[derive(ConstParamTy, PartialEq, Eq)]` if we detect all 3 are missing. Otherwise, users will add `#[derive(ConstParamTy)]` then get hit with follow-up errors, lol.

Reply:
How hard would it be to also suggest adding those derives in this error message? I think it would definitely be a better error if we did but I am fine to land this without doing that if its non-trivial